flood_adapt/adapter/direct_impacts_integrator.py

```diff
@@ -17,44 +17,72 @@
-from flood_adapt.config import Settings
-from flood_adapt.log import FloodAdaptLogging
+from flood_adapt.misc.config import Settings
+from flood_adapt.misc.log import FloodAdaptLogging
 from flood_adapt.object_model.direct_impact.impact_strategy import ImpactStrategy
-from flood_adapt.object_model.direct_impact.socio_economic_change import (
-    SocioEconomicChange,
+from flood_adapt.object_model.hazard.floodmap import FloodMap
+from flood_adapt.object_model.hazard.interface.models import Mode
+from flood_adapt.object_model.interface.database_user import DatabaseUser
+from flood_adapt.object_model.interface.path_builder import (
+    ObjectDir,
+    TopLevelDir,
+    db_path,
 )
-from flood_adapt.object_model.hazard.hazard import Hazard, ScenarioModel
+from flood_adapt.object_model.interface.projections import SocioEconomicChange
+from flood_adapt.object_model.interface.scenarios import ScenarioModel
 from flood_adapt.object_model.utils import cd
 
 
-class DirectImpacts:
+# TODO move code that is related to fiat to the Fiat Adapter
+class DirectImpacts(DatabaseUser):
     """All information related to the direct impacts of the scenario.
 
     Includes methods to run the impact model or check if it has already been run.
     """
 
+    logger = FloodAdaptLogging.getLogger(__name__)
     name: str
     socio_economic_change: SocioEconomicChange
     impact_strategy: ImpactStrategy
-    hazard: Hazard
-    has_run: bool = False
 
-    def __init__(self, scenario: ScenarioModel, database, results_path: Path) -> None:
-        self._logger = FloodAdaptLogging.getLogger(__name__)
+    def __init__(self, scenario: ScenarioModel) -> None:
         self.name = scenario.name
-        self.database = database
         self.scenario = scenario
-        self.results_path = results_path
+        self.site_info = self.database.site
+
         self.set_socio_economic_change(scenario.projection)
         self.set_impact_strategy(scenario.strategy)
-        self.set_hazard(scenario, database, self.results_path.joinpath("Flooding"))
-        # Get site config
-        self.site_toml_path = self.database.static_path / "site" / "site.toml"
-        self.site_info = database.site
-        # Define results path
-        self.impacts_path = self.results_path.joinpath("Impacts")
-        self.fiat_path = self.impacts_path.joinpath("fiat_model")
-        self.has_run = self.has_run_check()
+
+    @property
+    def hazard(self) -> FloodMap:
+        return FloodMap(self.name)
+
+    @property
+    def results_path(self) -> Path:
+        return db_path(
+            TopLevelDir.output, object_dir=ObjectDir.scenario, obj_name=self.name
+        )
+
+    @property
+    def impacts_path(self) -> Path:
+        return self.results_path / "Impacts"
+
+    @property
+    def fiat_path(self) -> Path:
+        return self.impacts_path / "fiat_model"
+
+    @property
+    def has_run(self) -> bool:
+        return self.has_run_check()
```

Comment:
Since this is in the integrator, you should see this as the controller class using the adapters to run scenarios. 
The idea is that by making them properties, we can have one of these instances doing more than one scenario since self.results_path is the scenario's output path in the db I think. 

I wouldnt mind putting all of these in __init__ tbh, but that is for a future PR as I want to stop changing stuff and get this one merged.